pkg/metainfo/kvmetainfo/objects.go

```diff
@@ -62,7 +62,45 @@ func (db *DB) GetObjectStream(ctx context.Context, bucket string, path storj.Pat
 // CreateObject creates an uploading object and returns an interface for uploading Object information
 func (db *DB) CreateObject(ctx context.Context, bucket string, path storj.Path, createInfo *storj.CreateObject) (object storj.MutableObject, err error) {
 	defer mon.Task()(&ctx)(&err)
-	return nil, errors.New("not implemented")
+
+	_, err = db.GetBucket(ctx, bucket)
+	if err != nil {
+		return nil, err
+	}
+
+	if path == "" {
+		return nil, storj.ErrNoPath.New("")
+	}
+
+	info := storj.Object{
+		Bucket: bucket,
+		Path: path,
+	}
+
+	if createInfo != nil {
+		info.Metadata = createInfo.Metadata
+		info.ContentType = createInfo.ContentType
+		info.Expires = createInfo.Expires
+		info.RedundancyScheme = createInfo.RedundancyScheme
+		info.EncryptionScheme = createInfo.EncryptionScheme
+	}
+
+	if info.ContentType == "" {
+		// TODO: autodetect content type from the path extension
+	}
+	
+	if info.RedundancyScheme == (storj.RedundancyScheme{}) {
+		// TODO: set default redundancy scheme from config
```

Comment:
@egonelbre How can we get the config settings for redundancy and encryption?

Reply:
Ask from the bucket.

Reply:
Do you mean to add fields like `DefaultCipher` and `DefaultRedundancy` to the `Bucket` type? Where will the bucket take the defaults if they are not explicitly specified during bucket creation?

Reply:
Yes, that makes most sense to me.

If bucket doesn't have it either 1. throw an error when creating the bucket or 2. use some sane defaults.

Reply:
If we go with "1. throw an error when creating the bucket" then it does not make sense to have `info` param in `CreateBucket` as pointer anymore. As passing `nil` would always result in error.

We can go with "2. use some sane defaults". But then the question is how to transfer the configuration from the `config.yaml` to the bucket creation?

The configuration is loaded here: https://github.com/storj/storj/blob/master/pkg/miniogw/config.go. Then the configuration is passed to the constructors like bucket store, stream store, etc. When we finish the migration this would be the place where we would create the `Metainfo` instead.

How about replacing `MetainfoLimits` with `MetainfoConfig` that would contain all relevant configs?

Reply:
I would prefer that the configuration/defaults are stored on the server. For 1. yes, then the method would change. With 2. I meant using sane constants that are defined in code, not read from config file.

Maybe for now, letting callers worry about passing in correct params for bucket would be easiest. We can always add a default behavior.

Reply:
OK, went for "2. use some sane defaults".